packages/front-end/components/Features/ScheduleInputs.tsx

```diff
@@ -3,50 +3,77 @@ import clsx from "clsx";
 import { format } from "date-fns";
 import { format as formatTimeZone } from "date-fns-tz";
 import React, { useEffect, useState } from "react";
+import { useUser } from "@/services/UserContext";
 import Field from "../Forms/Field";
 import SelectField from "../Forms/SelectField";
+import PremiumTooltip from "../Marketing/PremiumTooltip";
+import UpgradeMessage from "../Marketing/UpgradeMessage";
+import Toggle from "../Forms/Toggle";
 import styles from "./ScheduleInputs.module.scss";
 
 interface Props {
   defaultValue: ScheduleRule[];
   onChange: (value: ScheduleRule[]) => void;
+  setShowUpgradeModal: (value: boolean) => void;
 }
 
 export default function ScheduleInputs(props: Props) {
   const [rules, setRules] = useState(props.defaultValue);
   const [dateErrors, setDateErrors] = useState({});
+  const { hasCommercialFeature } = useUser();
+  const [toggleEnabled, setToggleEnabled] = useState(() => rules?.length > 0);
+
+  const canScheduleFeatureFlags = hasCommercialFeature("schedule-feature-flag");
 
   useEffect(() => {
     props.onChange(rules);
   }, [props, props.defaultValue, rules]);
 
+  useEffect(() => {
+    if (toggleEnabled && !rules.length) {
+      setRules([
+        {
+          enabled: true,
+          timestamp: null,
+        },
+        {
+          enabled: false,
+          timestamp: null,
+        },
+      ]);
+    } else if (toggleEnabled && rules.length) {
+      setRules(rules);
```

Comment:
I'm not clear on why we need to handle this case. Aren't initial rules already set in props defaultValue?

Reply:
Yes, but the defaultValue for `scheduleRules` for a new rule is an empty array, so when someone enables the toggle, we need to seed the scheduleRules with some base rules.

If we update the defaultValue to the base rules, We'll need to re-think how we define the initial state of `toggleEnabled`.

Reply:
I think the complexity here is coming from the fact that we're trying to use the toggle to modify the `scheduleRules`, but also using the `scheduleRules` to determine the value of the toggle. We can add another property that just controls the toggle's value, but I'm not sure if that's the better solution.

Reply:
Yeah definitely, I'd probably set the example rules in the defaultValue, and use a separate `toggleEnabled` var to show/hide the fields on the DOM and to keep/remove the schedule rules from the API's payload.